### PR TITLE
Use Abseil in FL demo

### PR DESCRIPTION
**Pull request by jvmancuso**
Switches the use of `argparse` to use `absl.flags` & `absl.app`, by TF convention. Also adds the ability to disambiguate the inner and outer learning rates when using the Reptile update rule.

### CODE REVIEW

File: examples/federated-learning/README.md [hunk 1]
```diff
@@ -80,7 +27,13 @@ python3 examples/federated-learning/run.py
 or remotely using:
 
 ```sh
-python3 examples/federated-learning/run.py config.json
+python3 examples/federated-learning/run.py --remote-config config.json
```
Comment (mortendahl) [inline examples/federated-learning/README.md]: should we rename `run.py` to `__main__.py`?

File: examples/federated-learning/run.py [hunk 2]
```diff
@@ -2,50 +2,56 @@
...
-import logging
 import argparse
+import logging
 
+from absl import app
+from absl import flags
 import tensorflow as tf
 import tf_encrypted as tfe
 
-parser = argparse.ArgumentParser(description='Process some integers.')
```
[the remark this answers is not in the document]
Comment (justin1121) [reply examples/federated-learning/run.py]: This it has to do with the players file where we're using `tfe.local_computation` its capturing the default Pond and ruining everything.

[the remark this answers is not in the document]
Comment (mortendahl) [reply examples/federated-learning/run.py]: yes I think you're right. how about moving the definition of `local_computation` out of Pond? nothing in it is Pond specific.

refactoring!

[the remark this answers is not in the document]
Comment (justin1121) [reply examples/federated-learning/run.py]: Makes sense, might be best in another PR though?

[the remark this answers is not in the document]
Comment (jvmancuso) [reply examples/federated-learning/run.py]: yeah I like this change -- I can do it here if it's really that straightforward!

[the remark this answers is not in the document]
Comment (jvmancuso) [reply examples/federated-learning/run.py]: looked into this and it's a bit less straightforward than just moving code, so let's do it in a separate PR :)

**Comment by jvmancuso** [comment]
@justin1121 any idea about the secure random test that's failing here?

**Comment by justin1121** [comment]
Yeah might just need to re-run the test?

**Pull request merged**